### ISSUE TITLE
Build Error For Support Libraries & dhis2-rule-engine

### ISSUE THREAD

**Issue by i-sourabh** [dhis2/dhis2-android-capture-app#98]
Getting Following error while I tried to build both 1.0.4 & 1.0.5

Unable to resolve dependency for ':app@debug/compileClasspath': Could not resolve project :dhis2-rule-engine.

However if I try to build without rule-engine it builds successfully but gives other compiles errors for support library variables.

**Comment by ahmedifhaam** [comment]
@Balcan i think you should check this out 

this was the solution to overcome the issue

https://github.com/Balcan/dhis2-rule-engine.git
clone this to another directory outside project
checkout android branch
copy all the files to the dhis2-rules-engine folder 
and then build


i think the submodule was not configured properly or the build instruction has to be written
[Documentation] https://github.com/dhis2/dhis2-android-capture-app/issues/87

**Comment by Balcan** [comment]
@i-sourabh @ahmedifhaam Yes, that's right. We imported the rule engine as a submodule in order to make some necesary corrections and add new functionality.
We are considering changing it back to a dependency as soon as all these changes have applied.

In further releases we will include build instructions to make it easy cloning/forking the repo.
Thank you!

**Comment by paleu256** [comment]
Just wondering if this was resolved or everyone has to do the work around

**Comment by paleu256** [comment]
Still cannot get a successful build even after following @ahmedifhaam workaround. 

> 
> 
> @Balcan i think you should check this out
> 
> this was the solution to overcome the issue
> 
> https://github.com/Balcan/dhis2-rule-engine.git
> clone this to another directory outside project
> checkout android branch
> copy all the files to the dhis2-rules-engine folder
> and then build
> 
> i think the submodule was not configured properly or the build instruction has to be written
> [Documentation] #87
> 
> git clone --recurse-submodules git://github.com/dhis2/dhis2-android-capture-app ??

I still cannot get a successful build even after doing this.

**Comment by paleu256** [comment]
@ahmedifhaam could you guide me further on how you were able to sort this out. I have followed your steps but I keep running into the same error 'Unable to resolve dependency for ':app@debug/compileClasspath': Could not resolve project :dhis2-rule-engine' 
I am probably missing something. This is what i've done
1. I've cloned the repo 'dhis2-android-capture-app' with the recursive option
2. Cloned the  repo 'dhis2-rule-engine' to a location outside the above project folder
3. checked out the android branch from within 'dhis2-rule-engine' repo
4. Copied the contents from 'dhis2-rule-engine' folder into to folder 'dhis2-rule-engine' located in my 'dhis2-android-capture-app' project.
5. clean the project
6. build
However, i get the same error.

**Comment by i-sourabh** [comment]
> @ahmedifhaam could you guide me further on how you were able to sort this out. I have followed your steps but I keep running into the same error 'Unable to resolve dependency for ':app@debug/compileClasspath': Could not resolve project :dhis2-rule-engine'
> I am probably missing something. This is what i've done
> 
> 1. I've cloned the repo 'dhis2-android-capture-app' with the recursive option
> 2. Cloned the  repo 'dhis2-rule-engine' to a location outside the above project folder
> 3. checked out the android branch from within 'dhis2-rule-engine' repo
> 4. Copied the contents from 'dhis2-rule-engine' folder into to folder 'dhis2-rule-engine' located in my 'dhis2-android-capture-app' project.
> 5. clean the project
> 6. build
>    However, i get the same error.

Hi @paleu256 , checkout the dhis2-rule-engine to '991a8eb1c2ae6516bad01929acd742f99b5e8226' and clone all the folders to main project's dhis2-rule-engine folder and try to build. It will build successfully without any errors.

**Comment by badda** [comment]
Are there any updated guidelines for cloning and building this project? It seems to be a recurring problem. We could do with some clone and build instructions on the project main page.
